thirdparty/git.go

```diff
@@ -210,3 +213,51 @@ func FormGitURLForApp(host, owner, repo, token string) string {
 
 	return fmt.Sprintf("https://%s/%s/%s.git", host, owner, repo)
 }
+
+// ParseGitVersion parses the git version number from the version string and returns a boolean indicating if it's an Apple Git version.
+func ParseGitVersion(version string) (string, bool, error) {
+	appleGitRegex := `(?: \(Apple Git-[\w\.]+\))?$`
+	matches := regexp.MustCompile(`^git version ` +
+		// capture the version major.minor(.patch(.build(.etc...)))
+		`(\w+(?:\.\w+)+)` +
+		// match and capture Apple git's addition to the version string
+		appleGitRegex,
+	).FindStringSubmatch(version)
+	if len(matches) != 2 {
+		return "", false, errors.Errorf("could not parse git version number from version string '%s'", version)
+	}
+
+	isAppleGit := strings.Contains(version, "Apple Git-")
+	return matches[1], isAppleGit, nil
+}
+
+// VersionMeetsMinimum checks if the version is less or equal to the minVersion.
```

Comment:
nit: could we reword to "returns true if the version is greater or equal to the minVersion" like you say in the test? This comment is a bit misleading for the equal case i think.

Reply:
done